### PR TITLE
DOC: Update lexsort docstring for axis kwargs

### LINKED ISSUE #10521: error in np.lexsort axis kwarg
Hello,

I would like to sort an array according to lexical order, sorting by first the second column. [np.lexsort](https://docs.scipy.org/doc/numpy/reference/generated/numpy.lexsort.html) is the function to use, but it seems not to handle the kwarg axis.

`>>> import numpy as np`
`>>> a = np.array([[0,1],[1,0],[0,0],[0,-1],[0,1],[1,-1]])`
`>>> a.ndim`
`2`
`>>> np.lexsort(a.T,axis=1)`
`Traceback (most recent call last):`
 `File "<stdin>", line 1, in <module>`
`ValueError: axis(=1) out of bounds`

Info version :
Debian Python 3.5.
`>>> np.__version__`
`'1.12.1'`

The same bug happens with Anaconda python and numpy.

The function `np.argsort` does not suffer from this problem, but is not what I want. In case, I could use `np.lexsort(a.T[::-1])` as a workaround.

### PR DESCRIPTION
**Pull request by liang3zy22**
[skip actions][skip cirrus][skip azp]

try to address the docstring part of #10521

### CODE REVIEW

**Comment by mdhaber** [comment]
I don't think this closes gh-10521; but with some adjustments, it would address the documentation part of the issue. I've taken the liberty of implementing my own suggestions for simplification. LMK if this still looks good @liang3zy22.

**Comment by liang3zy22** [comment]
It looks good, @mdhaber . It made the lexsort doc more clear. I think my change is a tutorial for how to use lexsort and the axis parameter.

**Comment by mattip** [comment]
Merging. Any more cleanup can be done in a follow-up. Thanks @liang3zy22 and @mdhaber this is a nice improvement.

File: numpy/_core/multiarray.py [hunk 1]
```diff
@@ -430,26 +430,26 @@ def lexsort(keys, axis=None):
...
+    sequence is used for the primary sort order, ties are broken by the
+    second-to-last key, and so on.
 
     Parameters
     ----------
-    keys : (k, N) array or tuple containing k (N,)-shaped sequences
-        The `k` different "columns" to be sorted.  The last column (or row if
-        `keys` is a 2D array) is the primary sort key.
+    keys : (k, m, n, ...) array-like
+        The `k` keys to be sorted. The *last* key (e.g, the last
```
Comment (mdhaber) [inline numpy/_core/multiarray.py:441]:
```suggestion
        The `k` keys to be sorted. The *last* key (e.g., the last
```

Comment (mattip) [reply numpy/_core/multiarray.py:441]: whoops

**Pull request merged**